search/routes/api/atom_extensions.py

```diff
@@ -0,0 +1,280 @@
+"""Classes derived from the Feedgen extension classes."""
+
+from typing import Any, Dict
+from feedgen.ext.base import BaseEntryExtension, BaseExtension
+from feedgen.entry import FeedEntry
+from feedgen.feed import FeedGenerator
+from lxml import etree
+
+
+class OpenSearchExtension(BaseExtension):
+    """Extension of the Feedgen base class to put OpenSearch metadata."""
+
+    def __init__(self: BaseExtension) -> None:
+        """Initialize extension parameters."""
+
+        self.__opensearch_totalResults = None
```

Comment:
I was following the conventions outlined in the `feedgen.ext` module. For example, [`feedgen.ext.podcast`](https://github.com/lkiesow/python-feedgen/blob/master/feedgen/ext/podcast.py):

```
class PodcastExtension(BaseExtension):
    '''FeedGenerator extension for podcasts.
    '''

    def __init__(self):
        # ITunes tags
        # http://www.apple.com/itunes/podcasts/specs.html#rss
        self.__itunes_author = None
        self.__itunes_block = None
        self.__itunes_category = None
        self.__itunes_image = None
        self.__itunes_explicit = None
        self.__itunes_complete = None
        self.__itunes_new_feed_url = None
        self.__itunes_owner = None
        self.__itunes_subtitle = None
        self.__itunes_summary = None
```

His implementations don't match standard Python usage - so the question is which paradigm takes precedence - language or library? I think that should be the language, as you do, so I could transform ours to use standard single-underscore syntax for private instance variables. 

Please confirm.

Reply:
It's totally fine to use the double-underscore, and library convention is a perfectly fine reason to do so. It's not a bad idea for a library that is built with extensive class hierarchies in mind (the `__` mangles the underlying attribute key behind the scenes so that it is specific to the class that sets it).

Reply:
Was mostly just curious